compiler/optimizations/gpuTransforms.cpp

```diff
@@ -733,11 +742,41 @@ bool GpuizableLoop::extractUpperBound() {
   return true;
 }
 
-void GpuizableLoop::reportNotGpuizable(const BaseAST* ast, const char *msg) {
+void GpuizableLoop::reportNotGpuizable(BaseAST* ast, const char *msg,
+                                       const std::vector<CallExpr*>* callStack) {
   this->reason = msg;
   if(this->compileTimeGpuAssertion_) {
     printNonGpuizableError(this->compileTimeGpuAssertion_, loop_);
-    USR_PRINT(ast, "%s", msg);
+
+    if (developer || ast->getModule()->modTag != MOD_INTERNAL) {
```

Comment:
What error message would we get for this case if you ran it without `--devel` (and assuming `module internal` were actually an internal module)?

```
module internal {
  var x = 1;

  proc f1() {
      x = 2;
  }

  proc f2() {
      f1();
  }
}

proc f3() {
    f2();
}

module user {
  on here.gpus[0] {
      @assertOnGpu
      foreach i in 1..100 {
          f3();
      }
  }
}
```

I'm thinking it would be something like this:

```
  user.chpl:5 error: Loop is marked with @assertOnGpu but is not eligible for execution on a GPU
  user.chpl:4: note:   reached via call to 'f3' in loop body here
```

Where we don't print `internal.chpl:xyz: note: call has outer var access` because `ast` is in an internal module but we still give the portion of the stack trace leading up to that, that falls in user code.

That seems confusing to me since it reads like the stack trace is telling you that the loop is reachable from a call in its body?